include/unifex/scheduler_concepts.hpp

```diff
@@ -100,14 +102,14 @@ namespace _schedule {
     }
     template <typename Scheduler>
     using _result_t =
-        typename decltype(_fn::_select<Scheduler>())::template apply<Scheduler>;
+        typename decltype(_impl::_select<Scheduler>())::template apply<Scheduler>;
   public:
     template(typename Scheduler)
       (requires _with_tag_invoke<Scheduler>)
     constexpr auto operator()(Scheduler&& s) const
         noexcept(is_nothrow_tag_invocable_v<_fn, Scheduler>)
         -> _result_t<Scheduler> {
-      return tag_invoke(_fn{}, static_cast<Scheduler&&>(s));
+      return tag_invoke(schedule, static_cast<Scheduler&&>(s));
```

Comment:
> Where is the `schedule` name defined now?

`<unifex/detail/unifex_fwd.hpp>`

> And why pass `schedule` here instead of, say `_impl{}`?

Because when people provide customizations for the `schedule`, they will be use `tag_t<schedule>`, not `_schedule::_impl`.